core/dbt/adapters/sql/connections.py

```diff
@@ -138,13 +141,13 @@ def data_type_code_to_name(cls, type_code: Union[int, str]) -> str:
         )
 
     def execute(
-        self, sql: str, auto_begin: bool = False, fetch: bool = False
+        self, sql: str, auto_begin: bool = False, fetch: bool = False, limit: Optional[int] = None
     ) -> Tuple[AdapterResponse, agate.Table]:
         sql = self._add_query_comment(sql)
         _, cursor = self.add_query(sql, auto_begin)
         response = self.get_response(cursor)
         if fetch:
-            table = self.get_result_from_cursor(cursor)
+            table = self.get_result_from_cursor(cursor, limit)
```

Comment:
because `limit` is only relevant if `fetch=true`, might it make more sense to have `fetch` become an `int` where:
- `0`: don't fetch i.e. `false`
- `-1`: fetch all
- `n`: fetch `n` rows

Reply:
I would be hesitant to change the type and function of `fetch`. This would certainly break our tests, which we can fix as part of the PR. But it would also break tests for a lot of community maintainers. Perhaps this is the right way to handle it, but my recommendation is to leave that as a 2.0 thing. Also, we would lose the option of a default `limit`; maybe not a big deal, but worth calling out.